File: PHP/Math.php
#! /usr/local/bin/php
<?php

$i = function($v){return $v;};

print <<<"Math"

これから数学演算を試します

進数変換
433045 = {$i(base_convert("433045",6,14))} (6進数 → 14進数)
	(2~36進数に変換可能)

dec2bin 138: {$i(   sprintf("%b",138)   )}
dec2oct 138: {$i(   sprintf("%o",138)   )}
dec2hex 138: {$i(   sprintf("%X",138)   )}

dec2bin 138: {$i(   decbin(138)   )}
dec2oct 138: {$i(   decoct(138)   )}
dec2hex 138: {$i(   dechex(138)   )}

bin2dec 1101111101: {$i(   bindec("1101111101")   )}
oct2dec       1575: {$i(   octdec("1575")         )}
hex2dec        37d: {$i(   hexdec("37d")          )}

定数
π = {$i(M_PI)}
e = {$i(M_E)}

符号
abs(+18) = {$i(   abs(+18)   )}
abs(-18) = {$i(   abs(-18)   )}

数学関数
sqrt(3)     = {$i(   sqrt(3)       )}
exp(1.145)  = {$i(   exp(1.145)    )}
log(23.14)  = {$i(   log(23.14)    )}
log10(1385) = {$i(   log10(1385)   )}
sin(π/3)    = {$i(   sin(M_PI/3)   )}
cos(π/3)    = {$i(   cos(M_PI/3)   )}
tan(π/3)    = {$i(   tan(M_PI/3)   )}
asin(1)     = {$i(   asin(1)       )}
acos(0.5)   = {$i(   acos(0.5)     )}
atan(-1)    = {$i(   atan(-1)      )}
atan2(12,5) = {$i(   atan2(12,5)   )}
hypot(12,5) = {$i(   hypot(12,5)   )} = (12²+5²) ^ ½
sinh(1.862) = {$i(   sinh(1.862)   )}
cosh(1.811) = {$i(   cosh(1.811)   )}
tanh(0.325) = {$i(   tanh(0.325)   )}

角度変換
rad2deg(1)   = {$i(   rad2deg(1)     )}
deg2rad(540) = {$i(   deg2rad(540)   )}

乱数
rand()       = {$i(   rand()         )} (0≦x≦getrandmax())
getrandmax() = {$i(   getrandmax()   )} (乱数範囲の最大値)

random_int() = {$i(   random_int(PHP_INT_MIN,PHP_INT_MAX)   )}

Math;
/*
	random_int() について

	• 暗号学的にrand()よりも安全な乱数を生成する
	• m≧PHP_INT_MIN, M≦PHP_INT_MAX として, random_int(m,M) と指定すると, m≤x≤M の範囲の整数を生成する
*/
print <<<"Math"

端数処理
floor(-3.14) = {$i(   floor(-3.14)   )} (小さい方の整数へ)
 ceil(-3.14) = {$i(    ceil(-3.14)   )} (大きい方の整数へ)
round(-3.14) = {$i(   round(-3.14)   )}
 (int)-3.14  = {$i(    (int)-3.14    )} (キャストは0に近い方へ)
Math;

print "\r\n\r\n";

?>

File: Fortran/Math.f90
subroutine Math

 real*16 pi
 real*16 e

 pi = acos(-1q0)
 e = exp(1q0)

 ! Fortranの数学関数は,入力値の精度によって出力値の精度が決定する
 ! ここでは4倍精度にしているので,戻り値も4倍精度になる。

 print *
 print '("これから数学演算を試します")'
 print *
 print '("進数変換")'
 print '("dec2bin 138: ",b8)',138
 print '("dec2oct 138: ",o8)',138
 print '("dec2hex 138: ",z8)',138
 print *
 print '("bin2dec 1101111101: ",i3)',b"1101111101"
 print '("oct2dec       1575: ",i3)',o"1575"
 print '("hex2dec        37d: ",i3)',z"37d"
 print *
 print '("定数")'
 print '("π = ",f35.33)',pi
 print '("e = ",f35.33)',e
 print *
 print '("最大/最小")'
 print '("max(3,4,6) = ",i1)',max(3,4,6)
 print '("min(3,4,6) = ",i1)',min(3,4,6)
 print *
 print '("符号")'
 print '("abs(+18) = ",i2)',abs(+18)
 print '("abs(-18) = ",i2)',abs(-18)
 print *
 print '("数学関数")'
 print '("sqrt(3q0)       = ",f35.33)',sqrt(3q0)
 print '("exp(1.145q0)    = ",f35.33)',exp(1.145q0)
 print '("log(23.14q0)    = ",f35.33)',log(23.14q0)
 print '("log10(1385q0)   = ",f35.33)',log10(1385q0)
 print '("sin(π/3)        = ",f35.33)',sin(pi/3q0)
 print '("cos(π/3)        = ",f35.33)',cos(pi/3q0)
 print '("tan(π/3)        = ",f35.33)',tan(pi/3q0)
 print '("asin(1q0)       = ",f35.33)',asin(1q0)
 print '("acos(0.5q0)     = ",f35.33)',acos(0.5q0)
 print '("atan(-1q0)      = ",f35.33)',atan(-1q0)
 print '("atan2(12q0,5q0) = ",f35.33)',atan2(12q0,5q0)
 print '("sinh(1.862q0)   = ",f35.33)',sinh(1.862q0)
 print '("cosh(1.811q0)   = ",f35.33)',cosh(1.811q0)
 print '("tanh(0.325q0)   = ",f35.33)',tanh(0.325q0)
 print *
 print '("複素数")'
 print '("aimag((3d0,4d0)) = ",f3.1)',aimag((3d0,4d0))
 print '("conjg((3d0,4d0)) = (",f3.1,",",f4.1,")")',conjg((3d0,4d0))
 print *

 print '("乱数")'

 ! 精度の低い乱数
 print '("ran(760013) = ",f8.6)',ran(760013)
 ! 0≤x<1 の範囲の単精度な均一分布の乱数を生成する
 ! 引数には,大きな値の奇数が入れなければならない

 ! 一般的に使用可能な乱数
 print '("rand(0) = ",f8.6)',rand(0)
 ! 0≤x<1 の範囲の単精度の乱数を生成する
 ! print '("drand(0) = ",f17.15)',drand(0)
 ! 0≤x<1 の範囲の倍精度の乱数を生成する
 print '("irand(0) = ",i10)',irand(0)
 ! 0≤x≤2147483647 の範囲の整数の乱数を生成する

 print *
 print '("端数処理")'
 print '("nint(-3.14) = ",i2," (四捨五入)")',nint(-3.14)
 print '("int(-3.14) = ",i2," (0に近い方の整数へ)")',int(-3.14)
 print '("ceiling(-3.14) = ",i2," (大きい方の整数へ)")',ceiling(-3.14)
 print *

end subroutine Math